### PR TITLE
Added repoadd wrapper

### PR DESCRIPTION
**Pull request by jdonszelmann**
I verified it working manually, and wrote some tests, but it's hard to fully test, and I haven't taken the time for that yet

### CODE REVIEW

**Comment by NULLx76** [comment]
If it all possible I'd rather use something like [go-alpm](https://github.com/Jguer/go-alpm) or was that not feasible?

File: pkg/repo_add/repoadd.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+// CommonOptions are options that are common to the repo-add and repo-remove commands.
+// Derived from https://www.archlinux.org/pacman/repo-add.8.html
+type CommonOptions struct {
+	// Generate a PGP signature file using GnuPG.
+	// This will execute gpg --detach-sign --use-agent on the generated database to generate
+	// a detached signature file, using the GPG agent if it is available.
+	// The signature file will be the entire filename of the database with a “.sig” extension.
+	// equivalent to `repo-add --sign`
+	Sign bool
```
Comment (NULLx76) [inline pkg/repo_add/repoadd.go:24]: Do we want to sign through repo-add or do the signing ourselves? As doing it throug repo-add means interfacing with the systems gpg which may or may not be what we want

File: pkg/repo_add/repoadd.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+	println(cmd.String())
+
+	err := cmd.Run()
+	if err != nil {
+		return err
+	}
+
+	if !cmd.ProcessState.Success() {
+		return fmt.Errorf("running repo-add failed: %d", cmd.ProcessState.ExitCode())
```
Comment (NULLx76) [inline pkg/repo_add/repoadd.go]: don't use `:` in error messages as that gets confusing when wrapping

File: pkg/repo_add/repoadd.go [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+	cmd := exec.Command(repoAddCommand, so...)
+	cmd.Args[0] = repoRemoveCommand
+
+	err := cmd.Run()
+	if err != nil {
+		return err
+	}
+
+	if !cmd.ProcessState.Success() {
+		return errors.New("running repo-remove failed")
```
Comment (NULLx76) [inline pkg/repo_add/repoadd.go]: Is it possible to also send stdout/stderr back to give a bit more context (applies to AddPackage as well)

File: pkg/repo_add/repoadd_test.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+	"testing"
+)
+
+func TestNewRepoAdd(t *testing.T) {
+	_, err := NewRepoAdd("test")
+
+	// If NewRepoAdd fails, the tests don't run on an arch system.
+	// By returning the test just passes on non Archlinux systems
+	if err != nil {
+		fmt.Printf("Not running on an archlinux system, marking test as passed.")
```
Comment (NULLx76) [inline pkg/repo_add/repoadd_test.go]: Skipping the test seems most appropriate?

File: pkg/repo_add/repoadd_test.go [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+		return
+	}
+}
+
+func TestIntegration(t *testing.T) {
+	dir := os.TempDir() + "/TestIntegration"
```
Comment (NULLx76) [inline pkg/repo_add/repoadd_test.go]: Maybe add `AAAA` to the dir as TestIntegration is a bit generic

File: pkg/repo_add/repoadd_test.go [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+	dbpath := dir + "/test.db.tar.gz"
+	pkgbuildpath := dir + "/PKGBUILD"
+
+	repo, err := NewRepoAdd(dbpath)
+
+	// If NewRepoAdd fails, the tests don't run on an arch system.
+	// By returning the test just passes on non Archlinux systems
+	if err != nil {
+		fmt.Printf("Not running on an archlinux system, marking test as passed.")
```
Comment (NULLx76) [inline pkg/repo_add/repoadd_test.go]: Again, passing when it is has failed seems weird, just skip?

**Review by NULLx76 — CHANGES_REQUESTED** [review]

File: pkg/repo_add/repoadd.go (hunk 1 above)
Comment (jdonszelmann) [reply pkg/repo_add/repoadd.go:24]: We probably want do do it ourselves. I didn't really include it for us to actually use it but mostly just to be complete and actually implement the entire command line interface

File: pkg/repo_add/repoadd.go (hunk 2 above)
Comment (jdonszelmann) [reply pkg/repo_add/repoadd.go]: Ill change it to put the error in parentheses

File: pkg/repo_add/repoadd.go (hunk 3 above)
Comment (jdonszelmann) [reply pkg/repo_add/repoadd.go]: Yeah, that's possible. Though I do remember that in apate we experienced that putting a pipe  on stdout in docker containers may not work as expected.

File: pkg/repo_add/repoadd_test.go (hunk 5 above)
Comment (jdonszelmann) [reply pkg/repo_add/repoadd_test.go]: Sure

File: pkg/repo_add/repoadd.go (hunk 3 above)
Comment (NULLx76) [reply pkg/repo_add/repoadd.go]: Yeah pipes don't work normally (as their isn't necessarily a stdout) but we can replace stdout with our own buffer iirc, we figured it out in apate so we can use that code as an example

Comment (jdonszelmann) [reply pkg/repo_add/repoadd.go]: yes I'll just take a look at what we did there. Expect a commit somewhere in the next 24 hours fixing it

File: pkg/repo_add/repoadd_test.go [hunk 7]
```diff
@@ -14,13 +13,12 @@ func TestNewRepoAdd(t *testing.T) {
 	// If NewRepoAdd fails, the tests don't run on an arch system.
 	// By returning the test just passes on non Archlinux systems
 	if err != nil {
-		fmt.Printf("Not running on an archlinux system, marking test as passed.")
-		return
+		t.Skip("Not running on an archlinux system, marking test as passed.")
```
Comment (NULLx76) [inline pkg/repo_add/repoadd_test.go]: Message isn't correct anymore

**Review by NULLx76 — APPROVED** [review]

**Pull request merged**